File: trainsim-client/src/models/SearchResult.ts
import Itinerary from "./Itinerary";

export default class SearchResult {
    private readonly _outboundItineraries: Itinerary[];
    private readonly _returnItineraries: Itinerary[];

    constructor(outboundItins: Itinerary[], returnItins: Itinerary[]) {
        this._outboundItineraries = outboundItins;
        this._returnItineraries = returnItins;
    }

    public get outboundItineraries(): Itinerary[] {
        return this._outboundItineraries;
    }

    public get returnItineraries(): Itinerary[] {
        return this._returnItineraries;
    }
}

File: trainsim-itinerary-microservice/src/main/java/edu/drexel/trainsim/itinerary/models/SearchResult.java
package edu.drexel.trainsim.itinerary.models;

import java.util.List;

public class SearchResult {
    private final List<Itinerary> outboundItineraries;
    private final List<Itinerary> returnItineraries;

    public SearchResult(List<Itinerary> outboundItins, List<Itinerary> returnItins) {
        outboundItineraries = outboundItins;
        returnItineraries = returnItins;
    }

    public List<Itinerary> getOutboundItineraries() {
        return outboundItineraries;
    }

    public List<Itinerary> getReturnItineraries() {
        return returnItineraries;
    }
}